internal/pkg/composable/providers/kubernetes/hints.go

```diff
@@ -256,19 +256,77 @@ func GetHintsMapping(k8sMapping map[string]interface{}, logger *logp.Logger, pre
 		composableMapping: mapstr.M{},
 		processors:        []mapstr.M{},
 	}
+	var hints mapstr.M
+	var containerProcessors []mapstr.M
 
 	if ann, ok := k8sMapping["annotations"]; ok {
 		annotations, _ := ann.(mapstr.M)
-		hints := utils.GenerateHints(annotations, "", prefix)
+
+		if containerEntries, err := annotations.GetValue(prefix + ".hints"); err == nil {
+			entries, ok := containerEntries.(mapstr.M)
+			if ok && len(entries) > 0 {
+				for key := range entries {
+					parts := strings.Split(key, "/")
+
+					if len(parts) > 1 {
+						if con, ok := k8sMapping["container"]; ok {
+							containers, ok := con.(mapstr.M)
+							if ok {
+								if cname, err := containers.GetValue("name"); err == nil {
+									if parts[0] == cname {
+										// If there are hints like co.elastic.hints.<container_name>/ then add the values after the / to the corresponding container
+										// Eg Annotation "co.elastic.hints.nginx/stream: stderr" will create a hints entry for container nginx
+										hints, containerProcessors = GenerateHintsForContainer(annotations, parts[0], prefix)
+									}
+								}
+							}
+						}
+					}
+				}
+			}
+		} else {
+			// If there are top level hints like co.elastic.hints/ then just add the values after the /
+			// Eg. Annotation "co.elastic.hints/stream: stderr" will will create a hints entries for all containers in the pod
+			hints = utils.GenerateHints(annotations, "", prefix)
+		}
+		logger.Debugf("Extracted hints are :%v", hints)
+
 		if len(hints) > 0 {
-			logger.Debugf("Extracted hints are :%v", hints)
-			hintData.composableMapping = GenerateHintsMapping(hints, k8sMapping, logger, cID)
-			logger.Debugf("Generated hints mappings are :%v", hintData.composableMapping)
+			hintData = GenerateHintsResult(hints, k8sMapping, annotations, logger, prefix, cID)
 
-			hintData.processors = utils.GetConfigs(annotations, prefix, processorhints)
-			logger.Debugf("Generated Processors are :%v", hintData.processors)
+			// Only if there are processors defined in a specific container we append them to the processors of the top level
+			if len(containerProcessors) > 0 {
+				hintData.processors = append(hintData.processors, containerProcessors...)
+			}
+			logger.Debugf("Generated Processors mapping :%v", hintData.processors)
 		}
+	}
+
+	return hintData
+}
+
+// Generates hints and processors list for specific containers
+func GenerateHintsForContainer(annotations mapstr.M, parts string, prefix string) (mapstr.M, []mapstr.M) {
```

Comment:
Just a small nit,  we could combine the last two arguments if they both are same. 
```suggestion
func GenerateHintsForContainer(annotations mapstr.M, parts, prefix string) (mapstr.M, []mapstr.M) {
```